### PR TITLE
feat: Controllers Code (#17)

### LINKED ISSUE #17: feat: Controllers Code
## Description of the feature
* The idea is to make the controller codes that are repeated stay here.

### PR DESCRIPTION
**Pull request by wsalles**
Avoiding repeated code from controllers.

closes #17

### CODE REVIEW

File: pkg/common/logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import (
+	"strings"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sirupsen/logrus"
+)
+
+// LoggerConfig defines a struct with required environment variables for logger
+type LoggerConfig struct {
```
Comment (renatoaguimaraes) [inline pkg/common/logger/logger.go:11]: Should implements the  `LoadFromEnvVars()` interface.

```
func (l * LoggerConfig) LoadFromEnvVars() error {
	return envconfig.Process("", l)
}
```

File: pkg/common/logger/logger.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	Type         string `envconfig:"LOG_TYPE" default:""`
+	CallerFirst  bool   `envconfig:"LOG_CALLER_FIRST" default:"false"`
+	TrimMessages bool   `envconfig:"LOG_TRIM_MESSAGES" default:"true"`
+	HideKeys     bool   `envconfig:"LOG_HIDE_KEYS" default:"false"`
+}
+
+// NewLogger initializes the logger
+func NewLogger() *logrus.Logger {
+	config := LoggerConfig{}
+	if err := envconfig.Process("", &config); err != nil {
```
Comment (renatoaguimaraes) [inline pkg/common/logger/logger.go:22]: We can use `InitConfigurationVariables(configs []ConfigurationVariables)` to initialize the logger config.

File: pkg/common/logger/logger.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		panic(err.Error())
+	}
+
+	loglevel, err := logrus.ParseLevel(config.Level)
+	if err != nil {
+		logrus.Errorf("Error: %v", err)
+		panic(err.Error())
```
Comment (renatoaguimaraes) [inline pkg/common/logger/logger.go:29]: We don't need to 'panic' here, just follow with the default log level.

**Review by renatoaguimaraes — CHANGES_REQUESTED** [review]

File: pkg/common/logger/logger.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	HideKeys     bool   `envconfig:"LOG_HIDE_KEYS" default:"false"`
+}
+
+// LoadFromEnvVars from the Logger
+func (c *Configuration) LoadFromEnvVars() error {
+	return envconfig.Process("", c)
+}
+
+// NewConfiguration for logger
+func NewConfiguration(level, output string, cf, trim, hk bool) *Configuration {
```
Comment (renatoaguimaraes) [inline pkg/common/logger/logger.go:25]: For this constructor, we don't need to pass these parameters, they come from environment variables.

File: pkg/common/logger/logger.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"strings"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sirupsen/logrus"
+)
+
+// Configuration defines a struct with required environment variables for logger
+type Configuration struct {
```
Comment (renatoaguimaraes) [inline pkg/common/logger/logger.go:11]: I think we can call it as `LoggingConfiguration` or `LoggerConfiguration`.

**Review by renatoaguimaraes — CHANGES_REQUESTED** [review]

File: pkg/common/logger/logger.go (hunk 5 above)
Comment (wsalles) [reply pkg/common/logger/logger.go:11]: according with Codacy, package logger and LoggerConfiguration causes stutter 🤣 
and then failed... we can rename the package from logger to logging or think like that

**Review by renatoaguimaraes — APPROVED** [review]

**Pull request merged**